src/main/java/com/intuit/graphql/orchestrator/schema/fold/UnifiedXtextGraphFolder.java

```diff
@@ -264,18 +295,49 @@ private TypeDefinition mergeSharedValueType(
                       .filter(fieldName -> newField.getName().equals(fieldName.getName()))
                       .findFirst();
 
-              if (!currentField.isPresent()) {
-                addNewFieldToObject(current, newField, newComerServiceProvider);
-              } else {
-                // To be used for inaccessible check later on
-              }
+                if (!currentField.isPresent()) {
+                    List<String> implementingTypesNotContainingField = (prexistingInfo.getTypes() != null) ? prexistingInfo.getTypes()
+                            .values()
+                            .stream()
+                            .filter(ObjectTypeDefinition.class::isInstance)
+                            .map(ObjectTypeDefinition.class::cast)
+                            .filter(objectTypeDefinition -> objectTypeImplementsInterface(objectTypeDefinition, newComer))
+                            .filter(implementingObjectType -> !objectTypeContainsFieldWithName(implementingObjectType, newField.getName()))
+                            .map(ObjectTypeDefinition::getName)
+                            .collect(Collectors.toList()) : new ArrayList<>();
+
+                    if(!implementingTypesNotContainingField.isEmpty()) {
```

Comment:
Do we need to check this? Shouldnt this be caught by the provider service itself